README.md

```diff
@@ -1,29 +1,200 @@
 # Validations with `form_tag`
 
-## Objectives
+Now that we've learned to handle the server side of validations, we need to
+take care of the client side.
+
+At this point, we'll be in step three of the following flow:
+
+1. User fills out the form and hits "Submit", transmitting the form data via
+   a POST request.
+2. The controller sees that validations have failed, and re-renders the form.
+3. **The view displays the errors to the user**.
+
+# Objectives
+
+After this lesson, you'll be able to...
 
-- render or redirect based on validation of instance in create/update
 - Prefill in form values based on an instance
 - Print out full error messages based on an invalid instance
-- introspect on errors for a field
-- apply an error class to a field conditionally based on errors on a field
+- Introspect on errors for a field
+- Apply an error class to invalid fields
+
+
+# Pre-Filling Form Values
+
+No one likes re-doing work. First, let's make sure we know how to pre-fill
+forms with the user's input so they don't have to type everything all over
+again.
+
+In the next lesson, `form_for` will be handling this for us, but `form_for`
+is *very* heavy on Rails magic, and continues to baffle scientists to this day,
+so we're going to do things by hand first, to understand what's really going on
+and what to do when tools like `form_for` don't work.
+
+Let's start with a vanilla form, using the [FormTagHelper][form_tag_helper]:
+
+[form_tag_helper]: http://api.rubyonrails.org/classes/ActionView/Helpers/FormTagHelper.html
+
+```erb
+<!-- app/views/people/new.html.erb //-->
+
+<%= form_tag("/people") do %>
+  <div class="field">
+    <%= label_tag "name", "Name" %>
+    <%= text_field_tag "name" %>
+  </div>
+  <div class="field">
+    <%= label_tag "email", "Email" %>
+    <%= text_field_tag "email", @person.email %>
+  </div>
+  <%= submit_tag "Create Person" %>
+<% end %>
+```
+
+Let's say we're working with this `Person` model:
+
+```ruby
+# app/models/person.rb
+
+class Person < ActiveRecord::Base
+  validates :name, format: { without: /[0-9]/, message: "does not allow numbers" }
+  validates :email, uniqueness: true
+end
+```
+
+This means validation will fail if we put numbers into the "Name" field, and
+the form will be re-rendered with the invalid `@person` object available.
+
+Now, let's plug the input back into the form:
+
+```erb
+<%= form_tag("/people") do %>
+  <div class="field">
+    <%= label_tag "name", "Name" %>
+    <%= text_field_tag "name", @person.name %>
+  </div>
+  <div class="field">
+    <%= label_tag "email", "Email" %>
+    <%= text_field_tag "email", @person.email %>
+  </div>
+  <%= submit_tag "Create" %>
+<% end %>
+```
+
+[text_field_tag]: http://api.rubyonrails.org/classes/ActionView/Helpers/FormTagHelper.html#method-i-text_field_tag
+
+As you can see from the [docs][text_field_tag], the second argument to
+`text_field_tag`, as with most form tag helpers, is the default value.
```

Comment:
Here again maybe give the HTML of this. At this point, I'm not sure what "default value" is.